### PR TITLE
Fix #226, Add Instructions on How to Squash and Amend Commits

### LINKED ISSUE #226: Add links or instructions to pull request squash and ammend section
**Checklist (Please check before submitting)**

* [x] I reviewed the [Contributing Guide](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md).
* [x] I reviewed the [cFS README.md file](https://github.com/nasa/cFS/blob/main/README.md) to see if the feature is in the major future work.
* [x] I performed a cursory search to see if the feature request is relevant, not redundant, nor in conflict with other tickets.

**Is your feature request related to a problem? Please describe.**
```.md
Once a Pull Request Ready for Review

    Verify the commit message and PR title use the template Fix #XYZ, SHORT_DESCRIPTION.
    Verify there is only one commit message. Squash or amend the commit messages if necessary.
    Verify that the PR passes all checks.
    The project team will label the issue and evaluate the pull request in the weekly configuration control board (CCB) meeting. For more information, visit The cFS CCB Process.
    If the pull request is accepted, it will be merged into cFS.

```

**Describe the solution you'd like**
Add links to instructions of "how" to squash a commit using git rebase or how to ammend a commit

**Describe alternatives you've considered**
Leave as is and address with users on a one-on-one, per PR basis

### PR DESCRIPTION
**Pull request by ArielSAdamsNASA**
**Checklist (Please check before submitting)**

* [x] I reviewed the [Contributing Guide](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md).
* [x] I signed and emailed the appropriate Contributor License agreement to GSFC-SoftwareRelease@mail.nasa.gov and copied cfs-program@lists.nasa.gov.

**Describe the contribution**
Fixes #226

- Added How to Squash Commits.
- Added How to Amend Commits.
- Added that users may use the format `Fix #XYZ, LONG_DESCRIPTION` for commit messages if necessary. 
- Added to create a commit message per topic and provided an example. 
- Added reference to https://chris.beams.io/posts/git-commit/

**Expected behavior changes**
Users should know how to squash and amend commits. 

**Contributor Info - All information REQUIRED for consideration of pull request**
Ariel Adams, ASRC Federal

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -90,12 +90,70 @@ Follow GitHub's fork-branch-pull request pattern.
 7. Provide your full name or GitHub username and your company or organization if applicable.
 
 #### Once a Pull Request Ready for Review
-1. Verify the commit message and PR title use the template `Fix #XYZ, SHORT_DESCRIPTION`.
-2. Verify there is only one commit message. Squash or amend the commit messages if necessary.
+1. Verify the commit message and PR title use the template `Fix #XYZ, SHORT_DESCRIPTION`. The commit message can use `Fix #XYZ, LONG_DESCRIPTION` if necessary.
```
Comment (skliper) [inline CONTRIBUTING.md:93]: Format is
```
Fix #XYZ, SHORT_DESCRIPTION

LONG_DESCRIPTION (optional)
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -90,12 +90,70 @@ Follow GitHub's fork-branch-pull request pattern.
...
+3. Squash or amend the commit messages if necessary. For more information, review the sections [How to Squash Commits](#how-to-squash-commits) and [How to Amend Commits](#how-to-amend-commits) .  
 3. Verify that the PR passes all checks.
-4. The project team will label the issue and evaluate the pull request in the weekly configuration control board (CCB) meeting. For more information, visit [The cFS CCB Process.](https://github.com/nasa/cFS/wiki/The-cFS-CCB-Process)
+4. The project team will label the issue and evaluate the pull request in the weekly configuration control board (CCB) meeting. For more information, visit [The cFS CCB Process](https://github.com/nasa/cFS/wiki/The-cFS-CCB-Process).
 5. If the pull request is accepted, it will be merged into cFS.
 
+#### How to Squash Commits
+1. Switch to the master branch and ensure you are up to date:
+```sh
+git checkout master && git pull
```
Comment (skliper) [inline CONTRIBUTING.md:103]: main

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -90,12 +90,70 @@ Follow GitHub's fork-branch-pull request pattern.
...
+1. Verify the commit message and PR title use the template `Fix #XYZ, SHORT_DESCRIPTION`. The commit message can use `Fix #XYZ, LONG_DESCRIPTION` if necessary. 
+2. Verify there is one commit message per topic. For example, review the [provided pull request](https://github.com/nasa/cFE/pull/1203/commits). 
+3. Squash or amend the commit messages if necessary. For more information, review the sections [How to Squash Commits](#how-to-squash-commits) and [How to Amend Commits](#how-to-amend-commits) .  
 3. Verify that the PR passes all checks.
-4. The project team will label the issue and evaluate the pull request in the weekly configuration control board (CCB) meeting. For more information, visit [The cFS CCB Process.](https://github.com/nasa/cFS/wiki/The-cFS-CCB-Process)
+4. The project team will label the issue and evaluate the pull request in the weekly configuration control board (CCB) meeting. For more information, visit [The cFS CCB Process](https://github.com/nasa/cFS/wiki/The-cFS-CCB-Process).
 5. If the pull request is accepted, it will be merged into cFS.
 
+#### How to Squash Commits
```
Comment (skliper) [inline CONTRIBUTING.md:100]: This seems to include both a rebase and a squash... do you want to address these separately?  There's also a few different methods, see https://github.com/nasa/cFS-GroundSystem/pull/137 (comments near the end from Joe and I).

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -90,12 +90,70 @@ Follow GitHub's fork-branch-pull request pattern.
...
+```
+4. A text editor will open with a file that lists all the commits in your branch, and in front of each commit is the word "pick". It looks something like this:
+```sh
+pick 1fc6c95 do something
+pick 6b2481b do something else
+pick dd1475d changed some things
+pick c619268 fixing typos
+```
+5. For every line except the first, you want to replace the word "pick" with the word "squash". It should end up looking like this:
+```sh
```
Comment (skliper) [inline CONTRIBUTING.md:121]: note f skips the step where you have to update the commit message (just uses the previous)

File: CONTRIBUTING.md (hunk 1 above)
Comment (ArielSAdamsNASA) [reply CONTRIBUTING.md]: Updated format in sections Ready to Add Your Code?, Once a Pull Request Ready for Review, and Soft Reset.

File: CONTRIBUTING.md (hunk 3 above)
Comment (ArielSAdamsNASA) [reply CONTRIBUTING.md:110]: Created three sections describing the three methods on how to squash commits. The sections are Interactive Rebase, Soft Reset, and Replace Branch.

File: CONTRIBUTING.md (hunk 4 above)
Comment (ArielSAdamsNASA) [reply CONTRIBUTING.md:134]: Added instructions for both pick and squash as well as r and f.

**Comment by astrogeco** [comment]
[CCB:2021-04-07](https://github.com/nasa/cFS/wiki/Community-CCB-Agenda:-April-07-2021) APPROVED
- Add reference to <https://chris.beams.io/posts/git-commit/>

**Comment by ArielSAdamsNASA** [comment]
> [CCB:2021-04-07](https://github.com/nasa/cFS/wiki/Community-CCB-Agenda:-April-07-2021) APPROVED
> 
> * Add reference to https://chris.beams.io/posts/git-commit/

Added reference under sections Ready to Add Your Code? and Once a Pull Request Ready for Review

**Review by astrogeco — APPROVED** [review]

**Pull request merged**